Recipely/Recipely/RecipesScreeen/CategoryScreen/Presenter/CategoryPresenter.swift

```diff
@@ -90,27 +105,102 @@ final class CategoryPresenter: CategoryPresenterProtocol {
+                sorted = category.sorted(by: { lhs, rhs in
+                    if lhs.foodKkal == rhs.foodKkal {
+                        return lhs.foodTime > rhs.foodTime
+                    }
+                    return lhs.foodKkal < rhs.foodKkal
+                })
+            } else {
+                sorted = category.sorted(by: { $0.foodKkal < $1.foodKkal })
+            }
+            view?.sortedRecip(recipe: sorted)
+        case (.caloriesHigh, _):
+            if sortedTime == .timeLow {
+                sorted = category.sorted(by: { lhs, rhs in
+                    if lhs.foodKkal == rhs.foodKkal {
+                        return lhs.foodTime > rhs.foodTime
+                    }
+                    return lhs.foodKkal > rhs.foodKkal
+                })
+            } else if sortedTime == .timeHigh {
+                sorted = category.sorted(by: { lhs, rhs in
+                    if lhs.foodKkal == rhs.foodKkal {
+                        return lhs.foodTime < rhs.foodTime
+                    }
+                    return lhs.foodKkal > rhs.foodKkal
+                })
+            } else {
+                sorted = category.sorted(by: { $0.foodKkal > $1.foodKkal })
+            }
+            view?.sortedRecip(recipe: sorted)
+        case (_, .timeLow):
+            sorted = category.sorted(by: { lhs, rhs in
+                if lhs.foodTime == rhs.foodTime {
+                    return lhs.foodKkal < rhs.foodKkal
+                }
+                return lhs.foodTime < rhs.foodTime
+            })
+            view?.sortedRecip(recipe: sorted)
+        case (_, .timeHigh):
+            sorted = category.sorted(by: { lhs, rhs in
+                if lhs.foodTime == rhs.foodTime {
+                    return lhs.foodKkal < rhs.foodKkal
+                }
+                return lhs.foodTime > rhs.foodTime
+            })
+            sorted = category.sorted(by: { $0.foodTime > $1.foodTime })
+            view?.sortedRecip(recipe: sorted)
         }
-        view?.updateButtonCalories(state)
     }
 
-    func updateSortingStateTimeButton() {
-        switch state {
-        case .none:
-            state = .up
-        case .up:
-            state = .down
-        case .down:
-            state = .none
+    func buttonTimeChange(category: [Recipes]) {
+        if sortedTime == .non {
+            sortedTime = .timeLow
+            view?.buttonTimeState(color: Constants.buttonPressedColor, image: Constants.stateImageTwo)
+            sortedRecipe(category: category)
+        } else if sortedTime == .timeLow {
+            view?.buttonTimeState(color: Constants.buttonPressedColor, image: Constants.stateImageThree)
+            sortedTime = .timeHigh
+            sortedRecipe(category: category)
+        } else if sortedTime == .timeHigh {
+            sortedTime = .non
+            view?.buttonTimeState(color: Constants.buttonDefaultColor, image: Constants.stateImageOne)
+            sortedRecipe(category: category)
```

Comment:
```suggestion
    func buttonTimeChange(category: [Recipes]) {
        if sortedTime == .non {
            sortedTime = .timeLow
            view?.buttonTimeState(color: Constants.buttonPressedColor, image: Constants.stateImageTwo)
        } else if sortedTime == .timeLow {
            view?.buttonTimeState(color: Constants.buttonPressedColor, image: Constants.stateImageThree)
            sortedTime = .timeHigh
        } else if sortedTime == .timeHigh {
            sortedTime = .non
            view?.buttonTimeState(color: Constants.buttonDefaultColor, image: Constants.stateImageOne)
            }
            sortedRecipe(category: category)
```
 и вместо if - switch